hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieBackedMetadata.java

```diff
@@ -1871,7 +1865,11 @@ private void testTableOperationsImpl(HoodieSparkEngineContext engineContext, Hoo
       validateMetadata(client);
 
       // Restore
-      client.restoreToInstant("20210101000600000", writeConfig.isMetadataTableEnabled());
+      if (metaClient.getTableType() == COPY_ON_WRITE) {
+        assertThrows(HoodieRestoreException.class, () -> client.restoreToInstant("20210101000600000", writeConfig.isMetadataTableEnabled()));
```

Comment:
@prashantwason : hey, can you help clarify, why do we expect this to fail just for COW table and not MOR ?

Reply:
found an issue with restore metadata update. i've fixed it and removed this conditional block from the test.